coriolis/tests/scheduler/test_scheduler_utils.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2023 Cloudbase Solutions Srl
+# All Rights Reserved.
+
+
+from unittest import mock
+
+from coriolis import constants
+from coriolis import exception
+from coriolis.scheduler import scheduler_utils
+from coriolis.tests import test_base
+
+
+class CoriolisTestException(Exception):
+    pass
+
+
+class SchedulerUtilsTestCase(test_base.CoriolisBaseTestCase):
+    """"Test suite for the Coriolis scheduler utils package."""
+
+    def setUp(self):
+        super(SchedulerUtilsTestCase, self).setUp()
+        self.scheduler_client = mock.MagicMock()
+        self.rpc_client_class = mock.MagicMock()
+        self.service = mock.MagicMock()
+        self.ctxt = mock.MagicMock()
+
+    def test_get_rpc_client_for_service(self):
+        scheduler_utils.RPC_TOPIC_TO_CLIENT_CLASS_MAP[
+            constants.WORKER_MAIN_MESSAGING_TOPIC] = self.rpc_client_class
+
+        self.service.topic = constants.WORKER_MAIN_MESSAGING_TOPIC
+        self.service.host = 'host'
+
+        result = scheduler_utils.get_rpc_client_for_service(self.service)
+
+        self.rpc_client_class.assert_called_once_with(
+            topic='coriolis_worker.host')
+
+        self.assertEqual(result, self.rpc_client_class.return_value)
+
+    def test_get_rpc_client_for_service_with_exception(self):
+        self.service.topic = 'non-existent-topic'
+        self.service.host = 'host'
+
+        self.assertRaises(exception.NotFound,
+                          scheduler_utils.get_rpc_client_for_service,
+                          self.service)
+
+    def test_get_any_worker_service_no_services(self):
+        self.scheduler_client.get_workers_for_specs.return_value = []
+
+        self.assertRaises(exception.NoWorkerServiceError,
+                          scheduler_utils.get_any_worker_service,
+                          self.scheduler_client, self.ctxt)
+
+    @mock.patch('coriolis.scheduler.scheduler_utils.db_api.get_service')
+    def test_get_any_worker_service_random_choice(self, get_service_mock):
+        service_mock1 = {'id': 'test_id1'}
+        service_mock2 = {'id': 'test_id2'}
+
+        self.scheduler_client.get_workers_for_specs.return_value = [
+            service_mock1, service_mock2]
+
+        get_service_mock.side_effect = [service_mock1, service_mock2]
+
+        result = scheduler_utils.get_any_worker_service(
+            self.scheduler_client, self.ctxt, random_choice=True)
+
+        self.assertIsInstance(result, dict)
+
+    @mock.patch('coriolis.scheduler.scheduler_utils.db_api.get_service')
+    def test_get_any_worker_service_raw_dict(self, get_service_mock):
+        service_mock = {'id': 'test_id'}
+
+        self.scheduler_client.get_workers_for_specs.return_value = [
+            service_mock]
+
+        get_service_mock.return_value = service_mock
```

Comment:
By the logic of the original method, it's the other way around. If `raw_dict=True`, `get_service_mock` should not have been called, so it's impossible to have its return value as result, but since you set the actual return value here, the test coincidentally passed.